### DISCUSSION TITLE
Use Webcam Script in Touchdesigner

### DISCUSSION THREAD

**Discussion by taygon** [roboflow/supervision#24] [Q&A]
I'm attempting to repurpose the webcam script from this [video](https://www.youtube.com/watch?v=QV85eYOb7gk) to use within Touchdesigner. What I'd like is to circumvent using cv2.VideoCapture, frame, etc. and instead convert a live webcam feed within Touchdesigner to a numpyArray and use that as my source.

Instead of using cv2.imshow, I'd then like to convert the final video output with the bounding boxes, confidence score etc. back to a numpyArray so that I can further manipulate it in Touchdesigner. Here's my script so far:

<img width="899" alt="script" src="https://user-images.githubusercontent.com/45796500/219786669-77f3b8fa-00fe-4b42-b460-dafefe053fc9.PNG">

The error I'm getting is: RuntimeError: Given groups=1, weight of size [16, 3, 3, 3], expected input[1, 4, 384, 640] to have 3 channels, but got 4 channels instead

There might be other things I'm doing wrong as well. Any help would be much appreciated.

**Comment by SkalskiP** [comment]
@taygon I'm not familiar with the tools you use, but could you try what is the shape of a frame? So print(frame.shape). I think that is the reason for the exception. Let me know what you find.

**Reply by taygon** [reply]
Hey thanks for the response.

When I print the shape I get (328, 640, 4)

**Reply by SkalskiP** [reply]
Yup. I’m not sure why it has 4 channels, but YOLOv8 takes RGB images - 3 channels.
